### ISSUE TITLE
mySql - JSON_VALUE and CHARACTER SET utf8mb4 results in parsing error

### ISSUE THREAD

**Issue by kzajaczkowski** [sqlfluff/sqlfluff#5832]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.
- There's one similar issue, but about `CREATE DATABASE`, not `JSON_VALUE`: https://github.com/sqlfluff/sqlfluff/issues/5754

### What Happened

Please have a look at the following code:
``` sql
SELECT
    JSON_VALUE('{"a":"b"}', '$.a' RETURNING CHAR(255) CHARACTER SET utf8mb4) AS JSON_VALUE_CHARSET_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_UNSIGNED_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_SIGNED_EXAMPLE_PARSABLE
UNION ALL
SELECT
    NULL AS JSON_VALUE_CHARSET_EXAMPLE_UNPARSABLE,
    JSON_VALUE('{"a":"1"}', '$.a' RETURNING UNSIGNED) AS JSON_VALUE_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_SIGNED_EXAMPLE_PARSABLE
UNION ALL
SELECT
    NULL AS JSON_VALUE_CHARSET_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_EXAMPLE_UNPARSABLE,
    JSON_VALUE('{"a":"1"}', '$.a' RETURNING SIGNED) AS JSON_VALUE_SIGNED_EXAMPLE_PARSABLE
```

### Expected Behaviour

There should be no `PRS` type linting errors as this is a valid mysql code.

### Observed Behaviour

Both `CHARACTER SET` and `UNSIGNED` give "unparsable section" error.
```
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models/....sql] FAIL                                                                                                                                                                               
L:   2 | P:  55 |  PRS | Line 2, Position 55: Found unparsable section:
                       | 'CHARACTER SET utf8mb4'
L:   8 | P:  45 |  PRS | Line 8, Position 45: Found unparsable section:
                       | 'UNSIGNED'
```

### How to reproduce

Lint the following code with dialect set to `mysql`:
``` sql
SELECT
    JSON_VALUE('{"a":"b"}', '$.a' RETURNING CHAR(255) CHARACTER SET utf8mb4) AS JSON_VALUE_CHARSET_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_UNSIGNED_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_SIGNED_EXAMPLE_PARSABLE
UNION ALL
SELECT
    NULL AS JSON_VALUE_CHARSET_EXAMPLE_UNPARSABLE,
    JSON_VALUE('{"a":"1"}', '$.a' RETURNING UNSIGNED) AS JSON_VALUE_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_SIGNED_EXAMPLE_PARSABLE
UNION ALL
SELECT
    NULL AS JSON_VALUE_CHARSET_EXAMPLE_UNPARSABLE,
    NULL AS JSON_VALUE_EXAMPLE_UNPARSABLE,
    JSON_VALUE('{"a":"1"}', '$.a' RETURNING SIGNED) AS JSON_VALUE_SIGNED_EXAMPLE_PARSABLE
```

### Dialect

mysql

### Version

<img width="258" alt="image" src="https://github.com/sqlfluff/sqlfluff/assets/65259263/25853166-be3d-40e5-aaf1-10b600c79ee0"><br/>

<img width="130" alt="image" src="https://github.com/sqlfluff/sqlfluff/assets/65259263/9ee58591-3773-46de-ba40-c717454308f3"><br/>
<img width="273" alt="image" src="https://github.com/sqlfluff/sqlfluff/assets/65259263/d3819f23-2bf4-4727-9201-47c1469faa8c">

### Configuration

Custom configuration not needed for this to occur

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kzajaczkowski** [comment]
I am happy to submit a PR, but currently am struggling in identifying what needs to be fixed as I am not familiar with the project. So any help much appreciated.
